inference-engine/src/cldnn_engine/ops/select.cpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "cldnn_program.h"
+#include "cldnn_common_utils.h"
+
+#include "ngraph/op/select.hpp"
+
+#include "api/select.hpp"
+#include "api/reorder.hpp"
+#include "api/reshape.hpp"
+
+namespace CLDNNPlugin {
+
+void CreateSelectOp(Program& p, const std::shared_ptr<ngraph::Node>& node) {
+    auto op = std::dynamic_pointer_cast<ngraph::op::v1::Select>(node);
+    if (!op)
+        THROW_IE_EXCEPTION << INVALID_OP_MESSAGE;
+
+    p.ValidateInputs(op, {3});
+    auto inputPrimitives = p.GetInputPrimitiveIDs(op);
+    std::string layerName = layer_type_name_ID(op);
+
+    auto outDims = op->get_output_shape(0);
+    auto outDimsN = outDims.size();
+
+    auto broadcast_type = op->get_auto_broadcast();
+
+    if (broadcast_type.m_type != ngraph::op::AutoBroadcastType::NONE &&
+        broadcast_type.m_type != ngraph::op::AutoBroadcastType::NUMPY) {
+        THROW_IE_EXCEPTION << "Unsupported broadcast type (" << broadcast_type.m_type << ") in layer " + op->get_friendly_name();
+    }
+
+    auto selectSpecificTensor = [](const InferenceEngine::SizeVector& dims, int def = 1) {
```

Comment:
This looks identical to CldnnTensorFromIEDims() code. Shouldn't we replace this with common call?

Reply:
yeah, replaced.